receiver/faroreceiver/faro.go

```diff
@@ -0,0 +1,128 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package faroreceiver // import "github.com/grafana/opentelemetry-collector-components/components/receiver/faroreceiver"
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"sync"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+)
+
+type faroReceiver struct {
+	cfg *Config
+	set receiver.CreateSettings
+
+	server       *http.Server
+	startOnce    sync.Once
+	shutdownOnce sync.Once
```

Comment:
We don’t typically promise your receiver will start and stop once. There might be times where it is reloaded.